src/pages/dataElements/form/fields.tsx

```diff
@@ -428,10 +609,19 @@ export function CategoryComboField() {
     )
 }
 
+/**
+ *
+ * OptionSet
+ *
+ */
 export function OptionSetField() {
     const newOptionSetLink = useHref('/optionSets/new')
-    const { input, meta } = useField('optionSet.id', {
+    // Not using a dot path because setting the value to an empty string
+    // removes the option set from the form state entirely
+    const { input, meta } = useField('optionSet', {
         validateFields: [],
+        format: (optionSet) => optionSet.id,
+        parse: (id) => ({ id }),
```

Comment:
I actually think this can be a good pattern - eg. keep the whole object/model and use format and parse. It could allow us to not have to fetch for dependent rules- and fetch more in one request in either form-load (initialize form) - or in specific fields. It could however make it a bit hard to follow what fields are actually there - so we would have to look at typings for it.

Reply:
> keep the whole object/model

That's already the case (the validation PR also uses the entire object. The form state is basically a replicate of the model).
I just noticed that when selecting an empty value, the entire object is thrown out of RFF's form state (it caused `values.optionSet.id` to throw as I was attempting to read `id` of `undefined`).

> It could however make it a bit hard to follow what fields are actually there - so we would have to look at typings for it.

I was thinking about this, too. To me this issue disappeared when I realized that I could simply look at the creation of the initial state as that's already the complete state (either with or without values, depending on whether it's the edit or new view)